packages/story-editor/src/app/pageDataUrls/pageDataUrlsProvider.js

```diff
@@ -40,94 +32,32 @@ import Context from './context';
 
 function PageDataUrlProvider({ children }) {
   const [dataUrls, setDataUrls] = useState({});
-  const taskQueue = useRef([]);
-  const isTaskQueueRunning = useRef(false);
-  const currentTask = useRef([null, null]);
+  const queueIdleTask = useIdleTaskQueue();
```

Comment:
I pretty much pulled out the IdleTaskQueue from this code so we could use it in other places. cut down on a lot of the code here, but I didn't alter the behavior of the Idle task queue at all, so most of that code hasn't really changed, just been moved around.

@swissspidy just pinging you here cus you had the suggestion initially and wanted to notify you that this is a util now available for consumption in other places